### ISSUE TITLE
vitest.config.mjs in 5.0 beta - Unknown Error: [object object]

### ISSUE THREAD

**Issue by axyl** [withastro/astro#12384]
### Astro Info

```block
Astro                    v5.0.0-beta.6
Node                     v22.9.0
System                   Linux (x64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             none
```

### Describe the Bug

With an _empty_ vitest.config.mjs 

`import { getViteConfig } from "astro/config";
export default getViteConfig({});`

Running tests gives a 
`Unhandled Error
Unknown Error: [object Object]`

Before being able to get to any tests.

I imagine something has changed with the configuration in version 5, but I can't spot anything on the changes list.

Switching to Astro 4.16.9 works fine.

### What's the expected result?

Expect to be able to run tests.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-zu7lpm?file=vitest.config.mjs&view=editor

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by apatel369** [comment]
I cannot reproduce this locally. It runs without issues as seen below:

![Screenshot](https://github.com/user-attachments/assets/0d4baad7-def7-4092-b5eb-74dcba365d40)

**Comment by joseluissb** [comment]
Having the same issue, node version 22.10

**Comment by apatel369** [comment]
It worked for me for node 22 as well. Trying to downgrade to the exact versions
```
Astro                    v5.0.0-beta.7
Node                     v22.11.0
System                   macOS (arm64)
Package Manager          pnpm
Output                   static
Adapter                  none
Integrations             none
```

**Comment by apatel369** [comment]
I was able to reproduce it by using exact versions.

![Screenshot 2024-11-08 at 10 17 58 PM](https://github.com/user-attachments/assets/f76f2602-2bce-43ce-842f-78772939aefc)

**Comment by HiDeoo** [comment]
Managed to temporarily workaround the issue on my end by overriding the Vite version used, e.g. with [`pnpm.overrides`](https://pnpm.io/package_json#pnpmoverrides):

```json
"pnpm": {
  "overrides": {
    "vitest>vite": "6.0.0-beta.6",
    "vite-node>vite": "6.0.0-beta.6"
  }
}
```

**Comment by joseluissb** [comment]
This has been fixed for me in `^5.0.0-beta.12`. I think this can be closed, thanks!

**Comment by delucis** [comment]
> vitest has reverted support for vite 6.0 in release [vitest-dev/vitest@`v2.1.7` (release)](https://github.com/vitest-dev/vitest/releases/tag/v2.1.7) I'm still having to use the overrides. I'm not sure if it is because of that reversion.

Yeah. You could also pin to 2.1.6 to avoid the overrides I think:

```json
"dependencies": {
  "vitest": "2.1.6"
}
```
